### PR TITLE
add demo for app doc

### PR DESCRIPTION
**Pull request by wangyikewxgm**
as title said

### CODE REVIEW

File: docs/en/application.md [hunk 1]
```diff
@@ -137,6 +137,31 @@ spec:
...
+           // +patchKey=name
+           spec: template: spec: containers: [parameter]
+        }
+        parameter: {
+           name: string
+           image: string
+           command?: [...string]
+        }
+```
+
```
Comment (wonderflow) [inline docs/en/application.md:164]: 不要直接贴这里。在下面这句话里加个链接

File: docs/en/application.md [hunk 2]
```diff
@@ -154,4 +179,151 @@ After the `Application` resource is applied to Kubernetes cluster, the KubeVela
 
 > TBD: the revision names and labels for resource instances are currently work in progress.
 
-> TBD: a demo for kubectl apply above Application CR and show full detailed underlying resources.
+## Verify generated resources
```
Comment (wonderflow) [inline docs/en/application.md:182]:
```suggestion
## Run Application

```

File: docs/en/application.md [hunk 3]
```diff
@@ -154,4 +179,151 @@ After the `Application` resource is applied to Kubernetes cluster, the KubeVela
...
+Apply these definitions and application yaml above.
```
Comment (wonderflow) [inline docs/en/application.md:183]:
```suggestion
Apply these definitions and application yaml above, then you'll get the application started.
```

File: docs/en/application.md [hunk 4]
```diff
@@ -154,4 +179,151 @@ After the `Application` resource is applied to Kubernetes cluster, the KubeVela
...
+
```
Comment (wonderflow) [inline docs/en/application.md:184]: 1. get the application and see the status
2. Just show the application by `kubectl get deploy` is fine, no more details needed.

File: docs/en/application.md [hunk 5]
```diff
@@ -154,4 +154,50 @@ After the `Application` resource is applied to Kubernetes cluster, the KubeVela
 
 > TBD: the revision names and labels for resource instances are currently work in progress.
 
-> TBD: a demo for kubectl apply above Application CR and show full detailed underlying resources.
+## Run Application
+You should make sure those workloads and traits definition installed already, you can find sidecar trait [here](../examples/registry/sidecar.yaml).
```
Comment (yangsoon) [inline docs/en/application.md:158]: please change `../examples/registry/sidecar.yaml`  to  `https://github.com/oam-dev/kubevela/tree/master/docs/examples/registry/sidecar.yaml`

**Review by wonderflow — APPROVED** [review]

File: docs/en/application.md [hunk 6]
```diff
@@ -154,4 +179,46 @@ After the `Application` resource is applied to Kubernetes cluster, the KubeVela
 
 > TBD: the revision names and labels for resource instances are currently work in progress.
 
-> TBD: a demo for kubectl apply above Application CR and show full detailed underlying resources.
+## Run Application
+Please should make sure those workloads and traits definition installed already.
```
Comment (wonderflow) [inline docs/en/application.md:183]:
```suggestion
```

**Pull request merged**